EIPS/eip-4760.md

```diff
@@ -0,0 +1,54 @@
+---
+eip: 4760
+title: SELFDESTRUCT bomb
+description: Deactivate SELFDESTRUCT by changing it to SENDALL, which does recover all funds to the caller but does not delete any contract code or storage. In order to give applications good time to switch to new constructions, do this via a stage of exponential gas cost increases.
+author: Guillaume Ballet (@gballet), Vitalik Buterin (@vbuterin), Dankrad Feist (@dankrad)
+discussions-to: https://github.com/ethereum/EIPs/issues/4761
+status: Draft
+type: Standards Track
+category: Core
+created: 2022-02-03
+requires: 
+---
+# SELFDESTRUCT bomb
+
+## Summary
+
+This EIP renames the `SELFDESCRUCT` opcode to `SENDALL`, and replaces its functionality. The new functionality will be only to send all Ether in the account to the caller.
+
+In order to give apps more warning even if their developers are completely unaware of the EIP process, this version will exponentially increase the gas costs of the opcode, so any developer has time to see this change and react by implementing a version of their contract that does not rely on `SELFDESTRUCT` .
+
+## Motivation
+
+The `SELFDESTRUCT` opcode requires large changes to the state of an account, in particular removing all code and storage. This will not be possible in the future with Verkle trees: Each account will be stored in many different account keys, which will not be obviously connected to the root account (see [Verkle tree EIP](https://notes.ethereum.org/5HDhQXstTaKtVqVbS7S9yw)).
+
+This EIP implements this change. Applications that only use SELFDESTRUCT to retrieve funds will still work.
+
+## Specification
+
+### Constants
+
+| Name | Value | Comment |
+|------|-------|---------|
+| `OLD_SELFDESTRUCT_COST` | 5000 | Current gas cost of `SELFDESTRUCT` opcode |
+| `HARD_FORK_BLOCK` | TBD | (Shanghai HF block height) |
+| `DOUBLING_SLOTS` | `2**16` | (Time for gas price to double, ca. 9 days) |
+| `DOUBLINGS_BEFORE_SENDALL` | `13` | `SELFDESTRUCT` will be converted to `SENDALL` at `HARD_FORK_BLOCK + DOUBLING_SLOTS * DOUBLINGS_BEFORE_SENDALL` |
+
+ * If `HARD_FORK_BLOCK <= slot < HARD_FORK_BLOCK + DOUBLING_SLOTS * DOUBLINGS_BEFORE_SENDALL`
+     * `SELFDESTRUCT` functionality remains unchanged
+     * `SELFDESTRUCT` gas cost is now `OLD_SELFDESTRUCT_COST * 2 ** ((slot - HARD_FORK_BLOCK) // DOUBLING_SLOTS)`
+ * For `slot >= HARD_FORK_BLOCK + DOUBLING_SLOTS * DOUBLINGS_BEFORE_SENDALL`
+     * The cost reverts back to `OLD_SELFDESTRUCT_COST`
+     * The `SELFDESTRUCT` opcode is renamed to `SENDALL`, and now only immediately moves all ETH in the account to the target; it no longer destroys code or storage or alters the nonce
+     * All refunds related to `SELFDESTRUCT` are removed
+
+## Rationale
+
+Getting rid of the `SELFDESTRUCT` opcode has been considered in the past (for arguments see [here](https://hackmd.io/@vbuterin/selfdestruct)), and there are currently no strong reasons to use it. Disabling it will be a requirement for statelessness.
```

Comment:
Remove external links.

This section should discuss why certain constants were chosen like 5000, 2**16, 13, etc.  The motivation section is the appropriate place to explain why this EIP in general is a good idea, Rationale is for explaining why specific decisions were made in this specification.